TESTING.md

```diff
@@ -33,10 +33,16 @@ These tests are in the `snapcraft/tests` directory.
 
 The integration tests are a group of suites that excercise snapcraft as a black box. They are only allowed to set up the environment where snapcraft runs and create files; but for the execution phase of the test they can only run the snapcraft command or one of its subcommands. To verify the results they can check the output printed to the command line, the return value of the snapcraft command, and any files created during the execution.
 
-This suite was split in three: plugins, store and other integration tests. This split is artificial, we made it just because the full suite takes more time than what Travis allows for a single job.
+This suite was split in four: plugins, store, snapd and other integration tests. This split is artificial, we made it just because the full suite takes more time than what Travis allows for a single job.
 
 These tests are in the `integration_tests` directory, with the `snapcraft.yamls` and other source files for the tests snaps in `integration_tests/snaps`.
 
+### Slow tests
+
+Some tests take too long. This affects the pull requests because we have to wait for a long time, and they will make Travis CI timeout because we have only 50 minutes per suite in there. The solution is to tag these tests as slow, and don't run them in all pull requests. These tests will only be run in autopkgtests.
+
+To mark a test as slow, set `slow_test = True` before the test `setUp`.
```

Comment:
all implementations in here set it as a class attribute instead of in `setUp`. While working it is confusing and potentially error prone.

Reply:
I think it makes it less error prone to put it as a class attribute, because that assignment will always run before setUp.

From your comment, it's not clear what's your preference. I could document that the slow_test = True must be set in a class attribute, which is not totally correct, but could make it easier to understand.